app/src/main/java/com/depoza/design/state/State.kt

```diff
@@ -1,15 +1,15 @@
 package com.depoza.design.state
 
-import android.support.annotation.DrawableRes
+import androidx.annotation.DrawableRes
 import java.io.Serializable
 
 /**
  * State description as UI component
  */
 data class State(
-    val tag: String,
-    @DrawableRes val imgResId: Int? = null,
-    val title: String? = null,
-    val subtitle: String? = null,
-    val buttons: List<Button>? = null
+    var tag: String,
```

Comment:
Could you explain the reason to make `State` as mutable?

Reply:
Because I wanted the state model to be modified later. To change without new references from memory and creating a new model .

Reply:
For such cases, you should create new one State object and apply it
```kotlin
val newState = // inisializing new state
stateView.apply(newState)
```

Reply:
Yes thats true but something like that was needed in my project. As i said if you create new state, keeps new space in memory, am i right? A very simple situation maybe not important as you wish.